File: python/proxy_pattenr/gumball_monitor.py
from gumball_machine import GumballMachine


class GumballMonitor:

    def __init__(self, machine: GumballMachine):
        self.machine = machine

    def report(self):
        print("위치:", self.machine.get_location())
        print("재고:", self.machine.get_count())
        print("상태:", self.machine.get_state())

File: java/src/Proxy/GumballMonitor.java
package Proxy;

import java.rmi.RemoteException;
import java.util.*;

public class GumballMonitor {
    GumballMachineRemote machine;

    public GumballMonitor(GumballMachineRemote machine) {
        this.machine = machine;
}

    public void report() {
        try {
            System.out.println("위치: " + this.machine.getLocation());
            System.out.println("재고: " + this.machine.getCount());
            System.out.println("상태: " + this.machine.getState());
        } catch (RemoteException e) {
            e.printStackTrace();
        }

    }
}
